tf2_ros/include/tf2_ros/buffer_interface.h

```diff
@@ -55,11 +55,10 @@ namespace tf2_ros
     return time_msg;
   }
 
-  inline const tf2::TimePoint& fromMsg(const builtin_interfaces::msg::Time & time_msg)
```

Comment:
@clalancette :+1:

The function you mentioned with "we get away with it" also needs to be changed:

```
const std::string& getFrameId(const geometry_msgs::Vector3Stamped& t) {return t.header.frame_id;}
```

It is trivial to use this function in a way which leads to accessing invalid memory:

```
const std::string & bad()
{
  geometry_msgs::Vector3Stamped t;
  return getFrameId(t);
}

const std::string & s = bad();
```
Trying to use `s`: :boom: